### PR TITLE
Migrate to TrinoFileSystem

### PR DESCRIPTION
**Pull request by pajaks**
## Description
Relates to https://github.com/trinodb/trino/issues/16020

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Comment by pajaks** [comment]
@ebyhr Can you run tests with secretes?

**Comment by ebyhr** [comment]
/test-with-secrets sha=5cebb1ea53fa38d4d195915274893d7dcdad7a58

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeTableWithCustomLocation.java [hunk 1]
```diff
@@ -63,18 +60,18 @@ public void testCreateAndDrop()
         String tableName = "test_create_and_drop" + randomNameSuffix();
         assertQuerySucceeds(format("CREATE TABLE %s AS SELECT 1 as val", tableName));
         Table table = metastore.getTable(SCHEMA, tableName).orElseThrow();
-        assertThat(table.getTableType()).isEqualTo(TableType.MANAGED_TABLE.name());
+        assertThat(table.getTableType()).isEqualTo(MANAGED_TABLE.name());
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeTableWithCustomLocation.java:63]: nit: This change is unrelated to the file system migration. Ideally, it should be a separated commit.

**Review by ebyhr — APPROVED** [review]
